core/src/main/java/io/confluent/rest/validation/JacksonMessageBodyProvider.java

```diff
@@ -40,10 +42,13 @@
 public class JacksonMessageBodyProvider extends JacksonJaxbJsonProvider {
 
   public JacksonMessageBodyProvider() {
-    setMapper(new ObjectMapper());
+	ObjectMapper mapper = new ObjectMapper();
+	mapper.enable(DeserializationFeature.FAIL_ON_READING_DUP_TREE_KEY);
+    setMapper(mapper);
   }
 
   public JacksonMessageBodyProvider(ObjectMapper mapper) {
+	mapper.enable(DeserializationFeature.FAIL_ON_READING_DUP_TREE_KEY);
```

Comment:
Sorry I was thinking having a single key per record to be produced is the requirement, as defined in the API reference page. Key should be uniquely identify a record. However, I think other REST function may also use this to parse their request, which may allow duplicated meta-data. I will change it back.